### PR TITLE
feat(layout): add norwegian language

### PR DESCRIPTION
**Pull request by Richard87**
Copied from Swedish, that is very similar 

Words sourced from https://wortschatz.uni-leipzig.de/de/download/Norwegian%20Bokm%C3%A5l

Skjermbilde 2024-02-26 kl  23 13 13
Skjermbilde 2024-02-26 kl  23 14 39

Seems all corpus downloaded are CC-BY licensend:
Skjermbilde 2024-02-26 kl  23 20 47

### CODE REVIEW

**Comment by frzsombor** [comment]
Nice work @Richard87, I also just created a pull request for Hungarian language. 

Reviewing your PR I would suggest you to manually double-check and fix some of your changes because your commits include additional `console.log` in `KeyboardSettings.tsx`, they overwrite to all `packages/keybr-phonetic-model/lib/generate/corpus/*.csv.gz` files (just commit your additions here) and I would also suggest you to review added lines and move them to keep an alphabetical order where possible (I'm referring to your edits in files like `packages/keybr-keyboard/lib/load.ts` and `packages/keybr-layout/lib/language.ts`). I'm just adding these because I know this helps maintainers to review and merge pull request faster, while keeping the repo and the code clean.

File: packages/keybr-content-words/lib/data/words-nb.json [hunk 1]
```diff
@@ -0,0 +1,3002 @@
...
+  "leverer",
+  "stand",
+  "Vår",
+  "bestemt",
+  "vekst",
+  "april",
+  "ti",
+  "tilsvarende",
+  "avtale",
+  "USA",
```
Comment (aradzie) [inline packages/keybr-content-words/lib/data/words-nb.json:822]: Yay!

File: packages/keybr-content-words/lib/data/words-nb.json [hunk 2]
```diff
@@ -0,0 +1,3002 @@
...
+  "skaffe",
+  "veier",
+  "kunde",
+  "manglende",
+  "motta",
+  "nettsiden",
+  "planlagt",
+  "rent",
+  "sesong",
+  "Facebook",
```
Comment (aradzie) [inline packages/keybr-content-words/lib/data/words-nb.json:1495]: OMG!

File: packages/keybr-content-words/lib/data/words-nb.json [hunk 3]
```diff
@@ -0,0 +1,3002 @@
...
+  "øvrige",
+  "Blant",
+  "modell",
+  "nyheter",
+  "valget",
+  "Håper",
+  "etablere",
+  "følelser",
+  "varm",
+  "NRK",
```
Comment (aradzie) [inline packages/keybr-content-words/lib/data/words-nb.json:1425]: An abbreviation which does not have vowels.

File: packages/keybr-content-words/lib/data/words-nb.json
Comment (aradzie) [inline packages/keybr-content-words/lib/data/words-nb.json:1]: Ah, I've seen this corpus before when I was adding the German language. See the discussion and my comments in #99.

The news articles are written in a particular style and are skewed toward particular topics, such as politics. This shifts the frequencies of words, and the final word list does not look like a "normal", everyday language. It usually contains a lot of names, such as country, politician and brand names. Also, it is not uncommon to see heavy usage of the English language mixed with your own language.

Over time I've developed my own approach to building word frequency dictionaries. I usually download a corpus of text from the [OpenSubtitles database](https://opus.nlpl.eu/OpenSubtitles/corpus/version/OpenSubtitles) and process it with [my own tools](https://github.com/aradzie/keybr.com-corpus). Then I give the result to a native speaker for them to remove all the profane, vulgar and obscene words. Then I include the reviewed result into keybr.

I keep the development of corpora in a [separate repository](https://github.com/aradzie/keybr.com-corpus).

If you wish, we can keep this list for the time being, and update it later.

Or we can work together to replace this list using the process described above.

**Review by aradzie — COMMENTED** [review]
Hi, thanks for your contribution! I am super excited to see more languages added to the project.

I am sorry it took me so long to review your PR. I believe we are in similar time zones, so from now I will be able to answer much more quickly.

Also, there have been a few changes in the repository that seem to be incompatible with your PR. You will probably have to rebase on the current master branch. Again, I am sorry for the trouble.

File: packages/keybr-content-words/lib/data/words-nb.json [hunk 4]
```diff
@@ -0,0 +1,3002 @@
...
+  "arbeidsplassen",
+  "besøker",
+  "deltatt",
+  "driften",
+  "finansiering",
+  "fjerde",
+  "forbundet",
+  "kjøring",
+  "løfte",
+  "m",
```
[the remark this answers is not in the document]
Comment (Richard87) [reply packages/keybr-content-words/lib/data/words-nb.json:2763]: thanks, not that I know of at least, removed :)

File: packages/keybr-content-words/lib/data/words-nb.json [hunk 5]
```diff
@@ -0,0 +1,3002 @@
...
+  "ferien",
+  "firma",
+  "forskningen",
+  "grei",
+  "grønn",
+  "kilometer",
+  "motstand",
+  "parter",
+  "raske",
+  "the",
```
[the remark this answers is not in the document]
Comment (Richard87) [reply packages/keybr-content-words/lib/data/words-nb.json:2915]: removed. 👍

File: packages/keybr-content-words/lib/data/words-nb.json [hunk 6]
```diff
@@ -0,0 +1,3002 @@
...
+  "myke",
+  "planter",
+  "professor",
+  "sammenhenger",
+  "servering",
+  "utgangen",
+  "øyeblikk",
+  "Fredrikstad",
+  "Hamar",
+  "Russland",
```
[the remark this answers is not in the document]
Comment (Richard87) [reply packages/keybr-content-words/lib/data/words-nb.json:2998]: `  "Fredrikstad",  "Hamar"` are proper real city names in norway (I'll remove russland)

File: packages/keybr-layout/lib/layout.ts [hunk 7]
```diff
@@ -516,6 +516,21 @@ export class Layout implements XEnumItem {
       Geometry.MATRIX,
     ),
   );
+  static readonly NB_NO = new Layout(
+    /* id= */ "no",
+    /* xid= */ 0x87,
+    /* name= */ "Norwegian",
+    /* family= */ LayoutFamily.QWERTY,
```
[the remark this answers is not in the document]
Comment (Richard87) [reply packages/keybr-layout/lib/layout.ts]: fixed :)

File: packages/keybr-content-words/lib/data/words-nb.json
Comment (Richard87) [reply packages/keybr-content-words/lib/data/words-nb.json:1]: Thanks for the awesome work and solid review! I have rebased the PR, sorted all the list, and reverted the change to the .csv.gz files :) 

(and removed the console.log line 🫣)

I dont have much time to work on the corpus, maybe the next norwegian coming around can create a better one)

**Comment by Richard87** [comment]
Current the tests are failing locally with these errors:

```
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
[@keybr/keyboard test] ERROR DETECTED

Running /opt/homebrew/bin/npm run test, pid: 27552

> @keybr/keyboard@0.0.0 test
> ava


  ✔ diacritics › get diacritic
  ✔ diacritics › combine diacritic

  Uncaught exception in lib/keyboard.test.ts

  Error

  › XEnum (/Users/richard/Projects/keybr.com/packages/keybr-lang/lib/enum.ts:95:15)
  › Object.<anonymous> (/Users/richard/Projects/keybr.com/packages/keybr-layout/lib/layout.ts:564:25)
  › load (/Users/richard/Projects/keybr.com/packages/tsl/index.js:27:12)
  › Object.<anonymous> (/Users/richard/Projects/keybr.com/packages/keybr-layout/lib/index.ts:4:1)

  Uncaught exception in lib/load.test.ts

  Error

  › XEnum (/Users/richard/Projects/keybr.com/packages/keybr-lang/lib/enum.ts:95:15)
  › Object.<anonymous> (/Users/richard/Projects/keybr.com/packages/keybr-layout/lib/layout.ts:564:25)
  › load (/Users/richard/Projects/keybr.com/packages/tsl/index.js:27:12)
  › Object.<anonymous> (/Users/richard/Projects/keybr.com/packages/keybr-layout/lib/index.ts:4:1)

  ✘ lib/keyboard.test.ts exited with a non-zero exit code: 1
  ✘ lib/load.test.ts exited with a non-zero exit code: 1
  ─

  2 tests passed
  2 uncaught exceptions
npm ERR! Lifecycle script `test` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: @keybr/keyboard@0.0.0 
npm ERR!   at location: /Users/richard/Projects/keybr.com/packages/keybr-keyboard 
Error: NPM Script Runner: /opt/homebrew/bin/npm run test exited with code 1

┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
Took a total of 9.06s to complete. 
➜  ~/Projects/keybr.com k:k8s-prod:happydogs-prod git:(norwegian) ✗ 
```

WIP

**Comment by aradzie** [comment]
I am away from my workstation are the moment, so I can't you right now. 

If you wish, I can merge the PR and fix the errors myself.

**Comment by aradzie** [comment]
Thanks! I'll update the website with your changes in a day or two.

**Comment by Richard87** [comment]
Wow, thanks 🙏  just testet it throughtly locally too, everything looks good!

also, super impressive project, it works *great*

**Pull request merged**